code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -0,0 +1,429 @@
+// ***************************************
+// *********** Place Warp Beacon
+// ***************************************
+/datum/action/xeno_action/place_warp_beacon
+	name = "Place Warp Beacon"
+	action_icon_state = "warp_beacon"
+	mechanics_text = "Binds our psychic essence to a spot of our choosing. We can use Hyperposition to swap locations with this essence."
+	plasma_cost = 150
+	cooldown_timer = 60 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_PLACE_WARP_BEACON
+
+/datum/action/xeno_action/place_warp_beacon/action_activate()
+	var/mob/living/carbon/xenomorph/wraith/TI = owner
+
+	TI.visible_message("<span class='xenowarning'>The air starts to violently roil and shimmer around [TI]!</span>", \
+	"<span class='xenodanger'>We begin to imprint our essence upon reality, causing the air about us to roil and shimmer...</span>") //Fluff
+
+	if(!do_after(TI, WRAITH_PLACE_WARP_BEACON_WINDUP, TRUE, TI, BUSY_ICON_BUILD)) //Channel time/wind up
+		TI.visible_message("<span class='xenowarning'>The space around [TI] abruptly stops shifting and wavering.</span>", \
+		"<span class='xenodanger'>We cease binding our essence to this place...</span>")
+		return fail_activate()
+
+	var/turf/T = get_turf(TI)
+	TI.visible_message("<span class='xenonotice'>\A shimmering point suddenly coalesces from the warped space around [T].</span>", \
+	"<span class='xenodanger'>We complete our work, binding our essence to this point.</span>", null, 5) //Fluff
+	var/obj/effect/alien/warp_beacon/WB = new(T) //Create the new beacon.
+	playsound(T, 'sound/weapons/emitter.ogg', 25, 1)
+	QDEL_NULL(TI.warp_beacon) //Delete the old beacon
+	TI.warp_beacon = WB //Set our warp beacon
+
+	succeed_activate()
+	add_cooldown()
+
+
+// ***************************************
+// *********** Hyperposition
+// ***************************************
+/datum/action/xeno_action/hyperposition
+	name = "Hyperposition"
+	action_icon_state = "hyperposition"
+	mechanics_text = "We teleport back to our warp beacon after a delay. The delay scales with the distance teleported."
+	plasma_cost = 25
+	cooldown_timer = 5 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_HYPERPOSITION
+
+
+/datum/action/xeno_action/hyperposition/action_activate()
+
+	var/mob/living/carbon/xenomorph/wraith/TI = owner
+	var/obj/effect/alien/warp_beacon/WB = TI.warp_beacon
+
+	if(!WB)
+		to_chat(TI, "<span class='xenodanger'>We have no warp beacon to teleport to!</span>")
```

Comment:
This needs a return because the code just continues.

Reply:
Done. Can't believe I overlooked that.